src/Entity/Reflection/PropertyMetadata.php

```diff
@@ -140,16 +140,38 @@ public function isValid(& $value)
 
 		foreach ($this->types as $type => $foo) {
 			if ($type === 'datetime') {
-				if ($value instanceof \DateTime || $value instanceof \DateTimeInterface) {
+				if ($value instanceof \DateTime) {
+
 					return TRUE;
-				}
+				} elseif ($value instanceof \DateTimeImmutable) {
+					$value = new \DateTime("@{$value->getTimestamp()}");
+
+					return TRUE;
+				} elseif (is_string($value) && $value !== '') {
+					$value = new \DateTime($value, new DateTimeZone(date_default_timezone_get()));
```

Comment:
this have probably different meaning! in the previous version the string could have the timezone offset, which have been changed later. in this case, it will not work (but I'm not sure which case will happen, but surely one timezone definition will be ignored.)